cmd/distrogen/yaml.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+
+	"gopkg.in/yaml.v3"
+)
+
+func yamlUnmarshalFromFile[T any](path string) (*T, error) {
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	var result T
+	err = yaml.Unmarshal(content, &result)
+	if err != nil {
+		return nil, wrapYamlErr(err, path)
```

Comment:
```suggestion
    if err != nil {
		return nil, fmt.Errorf("failed to unmarshal file %s something: %w", path, originalErr)
    }
    return &result, nil
```

Why not just this? I don't see `wrapYamlErr` used anywhere else.